src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,45 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import classNames from 'classnames';
 import { Loader } from '../Loader';
+import { Todo } from '../../types/Todo';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
+
+type Props = {
+  selectedTodo: Todo
+  onModalClose: () => void;
+};
+
+export const TodoModal: React.FC<Props> = ({
+  selectedTodo,
+  onModalClose,
+}) => {
+  const [user, setUser] = useState<User | null>(null);
+  const [isUserLoaded, setIsUserLoaded] = useState(false);
+  const {
+    id,
+    title,
+    completed,
+    userId,
+  } = selectedTodo;
+  // const {
+  //   name
+  //   email
+  // } = user
+
+  useEffect(() => {
+    getUser(userId)
+      .then(userFromServer => {
+        setUser(userFromServer);
+        setIsUserLoaded(true);
+      });
+  }, []);
```

Comment:
I'm not sure, but - why is this logic here? You choose todo and it's user on the main page, here you just show this result

Reply:
For me, this logic is easier and the user used  only in this component, so I don't see the need to lift state up